sdc/datatypes/hpat_pandas_dataframe_functions.py

```diff
@@ -31,65 +31,45 @@
 
 import operator
 import pandas
+import copy
+import numpy
+
+import sdc
 
 from numba import types
 from numba.extending import (overload, overload_method, overload_attribute)
+from sdc.hiframes.pd_dataframe_ext import DataFrameType
 from numba.errors import TypingError
 
-from sdc.datatypes.hpat_pandas_dataframe_types import DataFrameType
-from sdc.utils import sdc_overload_method
-
-
-@sdc_overload_method(DataFrameType, 'count')
-def sdc_pandas_dataframe_count(self, axis=0, level=None, numeric_only=False):
-    """
-    Pandas DataFrame method :meth:`pandas.DataFrame.count` implementation.
-
-    .. only:: developer
-
-        Test: python -m sdc.runtests sdc.tests.test_dataframe.TestDataFrame.test_count
-
-    Parameters
-    -----------
-    self: :class:`pandas.DataFrame`
-        input arg
-    axis:
-        *unsupported*
-    level:
-        *unsupported*
-    numeric_only:
-        *unsupported*
-
-    Returns
-    -------
-    :obj:`pandas.Series` or `pandas.DataFrame`
-            returns: For each column/row the number of non-NA/null entries. If level is specified returns a DataFrame.
-    """
-
-    _func_name = 'Method pandas.dataframe.count().'
-
-    if not isinstance(self, DataFrameType):
-        raise TypingError('{} The object must be a pandas.dataframe. Given: {}'.format(_func_name, self))
+from sdc.datatypes.hpat_pandas_series_functions import TypeChecker
 
-    if not (isinstance(axis, types.Omitted) or axis == 0):
-        raise TypingError("{} 'axis' unsupported. Given: {}".format(_func_name, axis))
+def sdc_pandas_dataframe_reduce_columns(df, name, params):
 
-    if not (isinstance(level, types.Omitted) or level is None):
-        raise TypingError("{} 'level' unsupported. Given: {}".format(_func_name, axis))
+    saved_columns = df.columns
+    data_args = tuple('data{}'.format(i) for i in range(len(saved_columns)))
+    all_params = ['df']
 
-    if not (isinstance(numeric_only, types.Omitted) or numeric_only is False):
-        raise TypingError("{} 'numeric_only' unsupported. Given: {}".format(_func_name, axis))
+    for key, value in params:
+        all_params.append('{}={}'.format(key, value))
+    ap = all_params.copy()
```

Comment:
I'm not sure the copy of all_params is needed. Also it would be good to have a comment here, to indicate that it relies on the fact we use series params to generate signature of DF method, i.e. something like:
```suggestion
    # This relies on parameters part of the signature of Series method called below being the same as for the corresponding DataFrame method
    series_call_params_str = '{}'.format(', '.join(all_params[1:]))
    func_definition = 'def _reduce_impl({}):'.format(', '.join(all_params))
```